tools/report-converter/tests/unit/test_smatch_parser.py

```diff
@@ -0,0 +1,79 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+"""
+This module tests the correctness of the SmatchAnalyzerResult, which
+used in sequence transform Smatch output to a plist file.
+"""
+
+
+import os
+import plistlib
+import shutil
+import tempfile
+import unittest
+
+
+from codechecker_report_converter.smatch.analyzer_result import \
+    SmatchAnalyzerResult
+
+
+class SmatchAnalyzerResultTestCase(unittest.TestCase):
+    """ Test the output of the SmatchAnalyzerResult. """
+
+    def setUp(self):
+        """ Setup the test. """
+        self.analyzer_result = SmatchAnalyzerResult()
+        self.cc_result_dir = tempfile.mkdtemp()
+        self.test_files = os.path.join(os.path.dirname(__file__),
+                                       'smatch_output_test_files')
+
+    def tearDown(self):
+        """ Clean temporary directory. """
+        shutil.rmtree(self.cc_result_dir)
+
+    def test_no_smatch_output_file(self):
+        """ Test transforming single C file. """
+        analyzer_result = os.path.join(self.test_files, 'files',
+                                       'sample.c')
+
+        ret = self.analyzer_result.transform(analyzer_result,
+                                             self.cc_result_dir)
+        self.assertFalse(ret)
+
+    def test_transform_dir(self):
+        """ Test transforming a directory. """
+        analyzer_result = os.path.join(self.test_files)
+
+        ret = self.analyzer_result.transform(analyzer_result,
+                                             self.cc_result_dir)
+        self.assertFalse(ret)
+
+    def test_transform_single_file(self):
+        """ Test transforming single output file. """
+        analyzer_result = os.path.join(self.test_files, 'sample.out')
+        self.analyzer_result.transform(analyzer_result, self.cc_result_dir)
+
+        plist_file = os.path.join(self.cc_result_dir,
+                                  'sample.c_smatch.plist')
+
+        with open(plist_file, mode='rb') as pfile:
+            res = plistlib.load(pfile)
+
+            # Use relative path for this test.
+            res['files'][0] = os.path.join('files', 'sample.c')
+
+            self.assertTrue(res['metadata']['generated_by']['version'])
+            res['metadata']['generated_by']['version'] = "x.y.z"
```

Comment:
I think it would be worth to check if the checker message and the positions are parsed and converted correctly.

Reply:
Agree.

Reply:
Does this mean something like this?

```python
self.assertTrue(res['diagnostics']['path']['message'])
self.assertTrue(res['diagnostics']['path']['location']['col'])
self.assertTrue(res['diagnostics']['path']['location']['file'])
self.assertTrue(res['diagnostics']['path']['location']['line'])
```

Or does it need to be checked with the fields of `sample.expected.plist` as well? Like for example:
```python
self.assertEqual(res['diagnostics']['path']['message'],exp['diagnostics']['path']['message'])
```
Cause the above stated is done for the whole plist at the end with:
```python
self.assertEqual(res, exp)
```

Reply:
My bad, I've checked the test again the `self.assertEqual(res, exp)` should be enough here.
We have some other parsing related tests which check the plist content similarly what you mentioned in the comment but that way of checking is not required here comparing the generated and the expected plist is enough.